resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/CallNotPermittedException.java

```diff
@@ -25,13 +25,23 @@
 public class CallNotPermittedException extends RuntimeException {
 
     /**
-     * The constructor with a CircuitBreaker.
+     * Static method to construct a {@link CallNotPermittedException} with a CircuitBreaker.
      *
      * @param circuitBreaker the CircuitBreaker.
      */
-    public CallNotPermittedException(CircuitBreaker circuitBreaker) {
-        super(String.format("CircuitBreaker '%s' is %s and does not permit further calls", circuitBreaker.getName(), circuitBreaker.getState()));
-    }
-}
+    public static CallNotPermittedException getCallNotPermittedException(CircuitBreaker circuitBreaker) {
+        boolean disableStackTraces = circuitBreaker.getCircuitBreakerConfig().isDisableStackTraces();
+
+        String message = String.format("CircuitBreaker '%s' is %s and does not permit further calls", circuitBreaker.getName(), circuitBreaker.getState());
 
+        if (disableStackTraces) {
```

Comment:
Can we use `return new CallNotPermittedException(message, disableStackTrace);` instead?